magpie/services.py

```diff
@@ -243,20 +243,26 @@ def user_requested(self):
         """
         Obtain the :term:`User` that was identified to obtain protected :term:`Resource` access.
         """
-        user = self.request.user
+        found = user = self.request.user
 
         # fallback lookup in case cookies exists but request method did not evaluate user
         # this can happen in situations where 'request.user' was pre-resolved when interacting with Twitcher
         if user is None:
             user = get_request_user(self.request)
-            self.request.user = user  # fix for future references using the expected location
 
         if not user:
             session = get_connected_session(self.request)
             anonymous = get_constant("MAGPIE_ANONYMOUS_USER", self.request)
             user = UserService.by_user_name(anonymous, db_session=session)
             if user is None:
                 raise RuntimeError("No Anonymous user in the database")
+        # 'request.user' expected None (and should remain as such) when anonymous (unauthenticated)
+        elif found is None and user is not None:
```

Comment:
The `and` second clause would always be true and thus useless since in the opposite the first `if` would kick in.

Reply:
I put it explicitly because I found it was not obvious otherwise (depending on previous if compared to comment mentioning following elif).